### ISSUE TITLE
cholesky/logdet/det gradients disagree with their own forward; the crate's own gradcheck rejects them

### ISSUE THREAD

**Issue by vyncint** [vyncint/oxmera#95]
## Summary

`cholesky`'s forward reads **only the lower triangle** — its own doc says "the upper is ignored, so a numerically asymmetric input is accepted". Its VJP returns Murray's **symmetrized** gradient `Ā = ½(S + Sᵀ)`. The two conventions do not match, so for `cholesky`, `logdet` and `det` the analytic gradient is:

- exactly **half** the true derivative on the strict lower triangle, and
- **nonzero where the true derivative is exactly zero** (the strict upper triangle, which the forward never reads).

Only the diagonal agrees. The crate's own `oxmera_autograd::gradcheck` — documented as "the tool that keeps every VJP honest" — **rejects these three ops** on a plain SPD input.

## Reproduction

```rust
// oxmera-tensor = "0.5.1", oxmera-autograd = "0.5.1"
use oxmera_tensor::Tensor;

fn main() {
    let vals = [4.0f32, 1.0, 0.5, 1.0, 3.0, 0.25, 0.5, 0.25, 2.0]; // symmetric, SPD
    let a = Tensor::from_slice(&vals, [3, 3]).unwrap().requires_grad_(true);
    a.logdet().unwrap().sum(&[]).unwrap().backward().unwrap();
    let an = a.grad().unwrap().to_vec_f32().unwrap();

    let fd = |i: usize| {                      // central difference of oxmera's OWN forward
        let (mut p, mut m, h) = (vals, vals, 1e-2);
        p[i] += h; m[i] -= h;
        let lp = Tensor::from_slice(&p, [3,3]).unwrap().logdet().unwrap().to_vec_f32().unwrap()[0];
        let lm = Tensor::from_slice(&m, [3,3]).unwrap().logdet().unwrap().to_vec_f32().unwrap()[0];
        (lp - lm) / (2.0 * h)
    };
    println!("(1,0) analytic={:.6} fd={:.6} ratio={:.4}", an[3], fd(3), an[3]/fd(3));
    println!("(0,1) analytic={:.6} fd={:.6}", an[1], fd(1));

    let b = Tensor::from_slice(&vals, [3, 3]).unwrap();
    println!("{:?}", oxmera_autograd::gradcheck(|xs: &[Tensor]| xs[0].logdet()?.sum(&[]), &[b], 1e-2, 3e-2));
}
```

## Observed

```
(1,0) analytic=-0.088235 fd=-0.176477 ratio=0.5000
(0,1) analytic=-0.088235 fd=0.000000
Err(invalid argument to gradcheck: input 0 element 1: analytic -0.08823529 vs numeric 0 (tol 0.03))
```

The same 0.5× / wrong-triangle pattern holds for `cholesky` and `det`.

## Expected

The analytic gradient agrees with a central finite difference of the crate's own forward — which is what `gradcheck` asserts, and what every other differentiable op does. (I gradient-checked the rest of the surface: unary, binary, reductions, softmax, matmul at several ranks, views, index ops, losses and nn layers all agree. These three are the exception.)

Either:
- return the one-sided gradient that matches the lower-triangle-only forward (zero above the diagonal), or
- make the forward read the full matrix and require symmetry the way `eigh` already does (`check_symmetric` + `EIGH_SYMMETRY_TOL`), so forward and backward agree on what the input is.

If the symmetric-perturbation convention is deliberate it needs to be stated where a user meets it — the `cholesky`/`logdet`/`det` rustdoc, README's linalg row ("batched `cholesky` (differentiable)", unqualified), and the Autograd section of `docs/LIMITATIONS.md`, which currently lists only `relu'(0) = 0` and the max-tie split as gradient conventions — along with the fact that `gradcheck` needs a symmetrizer for these ops.

Note `crates/oxmera-autograd/tests/gradcheck.rs::linalg_gradients_check` wraps every input in a symmetrizer, which is the one parameterisation under which the symmetrized gradient is correct. That is why the suite is green.

## Where

`crates/oxmera-tensor/src/cpu_linalg.rs` — `cholesky_backward`'s final `½(S + Sᵀ)` against the `cholesky` forward that reads only `i >= j`; wired up at `crates/oxmera-tensor/src/linalg.rs` and inherited by `logdet` and `det`.
